src/language-usability.rst

```diff
@@ -16,6 +16,193 @@ Usability of Type Systems
 API Usability
 =============
 
+..
+  Outline
+  * APIs are ubiquitous
+  * Examples
+  * High level goals of API
+  * Difficulty in achieving goals of API
+  * Distinction with language usability
+..
+
+Application Programming Interfaces (APIs) are interfaces that mediate interactions between software components.
+Examples include software libraries and frameworks, OS system calls, and web protocols.
+Nearly every program written will use an API :cite:`myers2016improving`.
+
+The goal of the API designer is to create an API that a broad range of
+programmers can use in their applications.
+This goal mirrors the goals of programming language designers --
+an API designer must a create a service that is "usable" to
+a broad and vaguely scoped audience.
+Like languages, APIs must be simple enough to be used
+by users who are unaware of the inner workings of the API.
+
+In many cases, both language and API design can be approached through a "bottom-up" manner -- designed with smaller components
+that can be combined to create complex behavior.
+However, in other cases, designing APIs involves involves putting a more usable layer atop a more complex existing API.
+This adds another layer of complexity in designing APIs.
+To illustrate this diffulty:
+
+    "The enormous size of public APIs contributes to these difficulties; for example, the Java Platform, Standard Edition API Specification includes more than 4,000 classes with more than 35,000 different methods, and Microsoft’s .NET Framework includes more than 140,000 classes, methods, properties, and fields."
+    :cite:`myers2016improving`.
+
+
+API Design Considerations
+-------------------------
+
+..
+  Outline
+  * Many aspects to design
+  * Also tradeoffs between aspects.
+    * Broad tradeoff - learnability and complexity
+  * Paper: Improving API usability work
+  * Paper: Automated measure paper
+..
+
+There are several design aspects to desiging usable APIs.
+A broad aspect of usability identified by both :cite:`myers2016improving`
+and :cite:`scheller2015automated` is the effect of *complexity* on usability.
+Changing the inherant complexity exhibits one of the many tradeoffs between design aspects of APIs.
+Making an API more powerful (allowing a user to do more things with the API)
+by increasing complexity decreases the learnability of the API.
+However, increasing the learnability of an API by lowering the complexity also lowers the power of the API.
+
+Myers et al. identify 6 key design aspects of API design, which we describe here.
```

Comment:
Introduce these things before the discussion on complexity above -- right now you talk about learnability and complexity (opposite of simplicity) before defining them.